src/hooks/use-throttled-state.tsx

```diff
@@ -0,0 +1,42 @@
+import { useCallback, useState, useRef, useMemo, useEffect } from 'react';
+
+import type { ThrottleSettings } from 'lodash';
+import throttle from 'lodash/throttle';
+
+export default function useThrottledState<State>(
+  initValue: State,
+  throttleMillis = 700,
+  throttleOptions: ThrottleSettings = {}
+) {
+  const { leading, trailing } = throttleOptions;
+  const stateRef = useRef(initValue);
+  const [, setState] = useState<State>(initValue);
+
+  const throttledSetState = useMemo(
```

Comment:
You can rename this to make it clearer that the setState here is triggering a rerender